geode-gfsh/src/test/java/org/apache/geode/management/internal/cli/commands/ClearCommandTest.java

```diff
@@ -0,0 +1,93 @@
+package org.apache.geode.management.internal.cli.commands;
+
+import static org.apache.geode.management.internal.i18n.CliStrings.CLEAR_REGION;
+import static org.apache.geode.management.internal.i18n.CliStrings.CLEAR_REGION_REGION_NAME;
+import static org.apache.geode.management.internal.cli.commands.ClearCommand.REGION_NOT_FOUND;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import org.apache.geode.cache.Cache;
+import org.apache.geode.cache.Region;
+import org.apache.geode.distributed.DistributedMember;
+import org.apache.geode.internal.cache.InternalCache;
+import org.apache.geode.management.internal.cli.domain.DataCommandRequest;
+import org.apache.geode.management.internal.cli.domain.DataCommandResult;
+import org.apache.geode.management.internal.cli.functions.DataCommandFunction;
+import org.apache.geode.management.internal.cli.result.model.ResultModel;
+import org.apache.geode.test.junit.rules.GfshParserRule;
+
+public class ClearCommandTest {
```

Comment:
This feels like more of an integration test as it's currently written, since it's not directly testing the ClearCommand class but rather how that class is used by gfsh. Instead of using `gfsh.executeAndAssertThat`, it might better to call `ClearCommand.clear()` directly and assert on the returned `ResultModel`.

It would also be good to verify that `callFunctionForRegion()` and `clearfn.remove()` are called with the expected/correct arguments, rater than just asserting the success/failure of the command, since otherwise the last two test cases end up being indistinguishable in terms of what they're asserting and the actual difference in behaviour between the two code paths isn't being verified.

Reply:
Originally I was calling clear directly but that introduce complications with the test needing to hardcode the parsing of ResultModel since we weren't using the gfsh tools that come with the gfsh rule. Ultimately I decided that using the gfsh rule and relying on the gfsh command line parsing to call the the actual method we're testing was the lesser of two evils in that regard. 

I have added the verifications, I agree they improve the quality of the test.